pep-9999.rst

```diff
@@ -0,0 +1,1176 @@
+     yes; after this PEP yes.
+
+   * Can it delete externally-installed packages? Currently
+     no, because pip has a special case for running in a
+     virtual environment and deleting packages outside it.
+     After this PEP, still no, in a more general way,
+     because system site packages will be outside any of the
+     ``sysconfig`` schemes for the environment.
+
+5. A distribution Python when used in a single-application
+   container image (e.g., a Docker container). In this use
+   case, the risk of breaking system software is lower,
+   since generally only a single application runs in the
+   container, and the impact is lower, since you can rebuild
+   the container and you don't have to struggle to recover a
+   running machine. There are also a large number of
+   existing Dockerfiles with an unqualified ``RUN pip
+   install ...`` statement, etc., and it would be good not
+   to break those.  So, builders of base container images
+   may want to ensure that the marker file is not present,
+   even if the underlying OS ships one by default. In such a
+   case:
+
+   * Can you ``pip install`` without an override? Currently
+     yes; after this PEP yes.
+
+   * Can it delete externally-installed packages? Currently
+     yes; after this PEP no. This is a behavior change which
+     we don't propose a way to override. However, since the
+     base image is generally minimal, there shouldn't be
+     much of a use case for simply uninstalling packages
+     (especially without using the distribution's own
+     tools). The common case is when pip wants to upgrade a
+     package, which previously would have deleted the old
+     version (except on Debian). After this change, the old
+     version will still be on disk, but pip will
+     still _shadow_ externally-installed packages, and we
+     believe this to be sufficient for this not to be a
+     breaking change in practice - a Python ``import``
+     statement will still get you the newly-installed
+     package.
+
+     If it becomes necessary to have a way to do this, we
+     suggest that the distribution should document a way for
+     the installer tool to access the ``sysconfig`` scheme
+     used by the distribution itself.  See the
+     `Recommendations for distributions`_ section for more
+     discussion.
+
+   It is the view of the authors of this PEP that it's still
+   a good idea to use virtual environments even in
+   single-application container images. For instance, the
+   example from `Docker's Python image`_ would be better
+   implemented as::
+
+       FROM python:3
+       WORKDIR /usr/src/app
+       RUN python3 -m venv venv
+       RUN venv/bin/pip install -U pip
+       COPY requirements.txt ./
+       RUN venv/bin/pip install --no-cache-dir -r requirements.txt
+       COPY . .
+       CMD ["venv/bin/python", "./your-daemon-or-script.py"]
+
+   .. _`Docker's Python image`: https://hub.docker.com/_/python
```

Comment:
Yes, I incorrectly thought that it was Debian Python because the image was based on Debian. Someone else mentioned this too but I lost track of that. I've removed this whole example.